demos/python_demos/multi_camera_multi_person_tracking/utils/analyzer.py

```diff
@@ -0,0 +1,222 @@
+        img = torch.Tensor(img.transpose(2, 0, 1)) / 255  # Scale
+        label_img = img.unsqueeze(0) if label_img is None else \
+            torch.cat((label_img, img.unsqueeze(0)))
+        return label_img
+
+    if use_images:
+        import torch
+    summary_writer = SummaryWriter(save_path)
+    embeddings_all = None
+    embeddings_avg = None
+    embeddings_clust = None
+    metadata_avg = []
+    metadata_all = []
+    metadata_clust = []
+    label_img_all = None
+    label_img_avg = None
+    label_img_clust = None
+    for i, sct in enumerate(scts):
+        for track in tqdm(sct.tracks, 'Processing embeddings: SCT#{}...'.format(i)):
+            # Collect average embeddings
+            if isinstance(track.f_avg.avg, int):
+                continue
+            embeddings_avg = track.f_avg.avg.reshape((1, -1)) if embeddings_avg is None else \
+                np.concatenate((embeddings_avg, track.f_avg.avg.reshape((1, -1))))
+            metadata_avg.append('sct_{}_'.format(i) + str(track.id))
+            if use_images:
+                label_img_avg = make_label_img(label_img_avg, track.crops[0])
+            # Collect all embeddings
+            features = None
+            offset = 0
+            for j, f in enumerate(track.features):
+                if f is None:
+                    offset += 1
+                    continue
+                features = f.reshape((1, -1)) if features is None else \
+                    np.concatenate((features, f.reshape((1, -1))))
+                metadata_all.append(track.id)
+                if use_images:
+                    crop = track.crops[j - offset]
+                    label_img_all = make_label_img(label_img_all, crop)
+            embeddings_all = features if embeddings_all is None else \
+                np.concatenate((embeddings_all, features))
+            # Collect clustered embeddings
+            for j, f_clust in enumerate(track.f_clust.clusters):
+                embeddings_clust = f_clust.reshape((1, -1)) if embeddings_clust is None else \
+                                    np.concatenate((embeddings_clust, f_clust.reshape((1, -1))))
+                metadata_clust.append(str(track.id))
+                if use_images:
+                    label_img_clust = make_label_img(label_img_clust, track.crops[j])
+
+    summary_writer.add_embedding(embeddings_all, metadata=metadata_all,
+                                 label_img=label_img_all, global_step=step, tag='All')
+    summary_writer.add_embedding(embeddings_avg, metadata=metadata_avg,
+                                 label_img=label_img_avg, global_step=step, tag='Average')
+    summary_writer.add_embedding(embeddings_clust, metadata=metadata_clust,
+                                 label_img=label_img_clust, global_step=step, tag='Clustered')
+    log.info('Embeddings have been saved successfully. To see the result use the next command: '
+             'tensorboard --logdir={}'.format(save_path))
```

Comment:
The `tensorboard` script is installed by the `tensorboard` package, while this module uses `tensorboardX`. Is it really necessary to install both?

Reply:
`tensorboard` does not provide `SummaryWriter` class that is available in `tensorboardX`. `tensorboard` can be used with `tensorflow` or `pytorch` while `tensorboardX` is a wrapper for `tensorboard` module and is used independently.

Reply:
So what you're saying is "yes". :-)